addons/survey/models/survey_survey.py

```diff
@@ -389,6 +427,25 @@ def _check_scoring_after_page_availability(self):
                 survey_names="\n- ".join(failing.mapped('title')))
             )
 
+    @api.constrains('user_id', 'restrict_user_ids')
+    def _check_survey_responsible_access(self):
+        """ When:
+                - a survey access is restricted to a list of users
+                - and there is a survey responsible,
+                - and this responsible is not survey manager (just survey officer),
+            check the responsible is part of the list."""
+        failing_survey_values = []
+        for survey_sudo in self.filtered(lambda s: bool(s.user_id - s.restrict_user_ids)).sudo():
+            try:
+                survey_sudo._compute_is_survey_restriction_allowed()
+                survey_sudo.with_user(survey_sudo.user_id).check_access_rule("write")
```

Comment:
You should generally invalidate the cache instead of triggering the compute manually; but it shouldn't be necessary as we should have updated the dependencies? (see other comments)